spec/tools/fix_auth/models/fix_configuration_script_spec.rb

```diff
@@ -0,0 +1,47 @@
+$LOAD_PATH << Rails.root.join("tools").to_s
+
+require "fix_auth"
+
+RSpec.describe FixAuth::FixConfigurationScript do
+  let!(:configuration_script) { FactoryBot.create(:configuration_script, :credentials => credentials) }
+  let(:legacy_key) { ManageIQ::Password::Key.new }
+  let(:pass)       { "password" }
+  let(:enc_old)    { ManageIQ::Password.encrypt(pass, legacy_key) }
+  let(:options)    { {:legacy_key => legacy_key, :silent => true} }
+
+  context "with nil credentials" do
+    let(:credentials) { nil }
+
+    it "does nothing" do
+      FixAuth::FixConfigurationScript.run(options)
+      expect(configuration_script.credentials).to be_nil
+    end
+  end
+
+  context "with no v2 encrypted passwords in credentials" do
+    let(:credentials) { {"foo" => "bar"} }
```

Comment:
Example of an unencrypted simple-string

Reply:
last rev I promise

1. It was blowing up when verbose was set to true
2. Wanted to make sure there was at least one encrypted field in the record (so it was a `contender` that then skipped encryting fields.

```diff
diff --git a/spec/tools/fix_auth/models/fix_configuration_script_spec.rb b/spec/tools/fix_auth/models/fix_configuration_script_spec.rb
index 10c2d885c2..eb5093dd1d 100644
--- a/spec/tools/fix_auth/models/fix_configuration_script_spec.rb
+++ b/spec/tools/fix_auth/models/fix_configuration_script_spec.rb
@@ -37,11 +37,17 @@ RSpec.describe FixAuth::FixConfigurationScript do
   end

   context "with v2 encrypted passwords in credentials" do
-    let(:credentials) { {"foo" => enc_old} }
+    let(:credentials) { {"foo" => enc_old, "foo2" => enc_old, "bar" => "other"} }

     it "re-encrypts the passwords" do
       FixAuth::FixConfigurationScript.run(options)
       expect(configuration_script.reload.credentials["foo"]).to be_encrypted(pass)
+      expect(configuration_script.reload.credentials["foo2"]).to be_encrypted(pass)
+    end
+
+    it "does nothing" do
+      FixAuth::FixConfigurationScript.run(options)
+      expect(configuration_script.reload.credentials["bar"]).to eq("other")
     end
   end
 end
diff --git a/tools/fix_auth/auth_config_model.rb b/tools/fix_auth/auth_config_model.rb
index a921989941..bc4515b153 100644
--- a/tools/fix_auth/auth_config_model.rb
+++ b/tools/fix_auth/auth_config_model.rb
@@ -15,7 +15,9 @@ module FixAuth

       def display_column(r, column, options)
         puts "    #{column}:"
-        traverse_column([], YAML.load(r.send(column)), options)
+
+        hash = r.send(column).kind_of?(Hash) ? r.send(column) : YAML.load(r.send(column))
+        traverse_column([], hash, options)
       end

       def password_field?(key)
```